lms/lmsdb/models.py

```diff
@@ -330,13 +330,21 @@ def create_hash(content: Union[str, bytes], *args, **kwargs) -> str:
 
     @classmethod
     def is_duplicate(
-            cls, content: Union[str, bytes], user: User, *,
-            already_hashed: bool = False,
+            cls, content: Union[str, bytes], user: User, exercise: Exercise,
+            *, already_hashed: bool = False,
     ) -> bool:
+
         hash_ = cls.create_hash(content) if not already_hashed else content
+
+        last_submission = cls.select(fn.MAX(cls.submission_timestamp)).where(
+            cls.solver == user,
+            cls.exercise == exercise,
+        )
+
         return cls.select().where(
```

Comment:
That's some great work you've done here, but I think i would prefer to phrase it as one query with `GROUP BY` and `HAVING`.